sql/src/main/java/io/crate/metadata/settings/session/SessionSettingRegistry.java

```diff
@@ -42,7 +45,14 @@
                 } else {
                     context.setDefaultSchema(null);
                 }
-            }).build();
+            })
+            .put(SEMI_JOIN_KEY, (parameters, expressions, context) -> {
+                if (expressions.size() == 1) {
+                    String value = ExpressionToStringVisitor.convert(expressions.get(0), parameters);
+                    context.setSemiJoinsRewriteEnabled(Boolean.valueOf(value));
```

Comment:
`Boolean.valueOf` is very lenient (everything not "true" is false). I think it would be better to use the `ExpressionToObjectVisitor` and either cast directly to boolean (only accepting BooleanLiterals) or use BooleanType.value to do a conversion - which would have the same semantic then as if it's a auto-cast in a regular expression.